src/core_plugins/kibana/public/dashboard/index.js

```diff
@@ -275,6 +276,16 @@ app.directive('dashboardApp', function (Notifier, courier, AppState, timefilter,
         $state.panels.push(createPanelState(hit.id, 'visualization', getMaxPanelIndex()));
       };
 
+      if ($route.current.params && $route.current.params[DashboardConsts.ADD_VIS_PARAM]) {
+        $scope.addVis({ id: $route.current.params[DashboardConsts.ADD_VIS_PARAM] });
+        kbnUrl.removeParam(DashboardConsts.ADD_TO_DASH_PARAM);
+        kbnUrl.removeParam(DashboardConsts.ADD_VIS_PARAM);
+      }
+
+      $scope.addNewVis = function () {
```

Comment:
Might as well make this a named function so it's easier to identify in the call stack:

```javascript
$scope.addNewVis = function addNewVis() {
```